build_for_ultimaker.sh

```diff
@@ -43,13 +43,94 @@ DEB_DIR=`pwd`/debian
 KCONFIG=`pwd`/configs/${BUILDCONFIG}_config
 KERNEL_BUILD=`pwd`/_build_armhf/${BUILDCONFIG}-linux
 
+INITRAMFS_COMPRESSION="${INITRAMFS_COMPRESSION:-.lzo}"
+INITRAMFS_ROOT_GID=${INITRAMFS_ROOT_GID:-0}
+INITRAMFS_ROOT_UID=${INITRAMFS_ROOT_UID:-0}
+INITRAMFS_SOURCE="${INITRAMFS_SOURCE:-initramfs/initramfs.lst}"
+INITRAMFS_IMG="${KERNEL_BUILD}/initramfs.cpio${INITRAMFS_COMPRESSION}"
+
+GEN_INIT_CPIO="${KERNEL_BUILD}/usr/gen_init_cpio"
+GEN_INITRAMFS_LIST="${KERNEL}/scripts/gen_initramfs_list.sh"
+
+BB_PKG="http://ftp.nl.debian.org/debian/pool/main/b/busybox/busybox-static_1.22.0-19+b3_armhf.deb"
+BB_BIN="busybox"
+
 # Set the release version if it's not passed to the script
 RELEASE_VERSION=${RELEASE_VERSION:-9999.99.99}
 
 # Initialize repositories
 git submodule init
 git submodule update
 
+busybox_get()
+{
+    local BB_DIR=$(mktemp -d)
+    local BB_AR="${BB_DIR}/busybox-static_armhf.deb"
+    local INITRAMFS_DIR="${1}"
+    local cwd=$(pwd)
+
+    if [ ! -d "${INITRAMFS_DIR}" ]; then
+        echo "No initramfs dir set to download busybox into."
+        exit 1
+    fi
+
+    if [ -z "${BB_DIR}" ]; then
+        echo "Unable to create temporary directory to get busybox."
+        exit 1
+    fi
+
+    wget -q "${BB_PKG}" -O "${BB_AR}"
```

Comment:
Faulthandling missing: wget can fail... timeouts and stuff.

Reply:
yep, as can the steps below :) but a) the script used set -e already to exit on error, and b) when the wget step fails, the step below will fail as well, where we check if we have an executable busybox binary :)